### ISSUE TITLE
Public network libvirt__iface_name ignored

### ISSUE THREAD

**Issue by lengers** [vagrant-libvirt/vagrant-libvirt#1704]
**Describe the bug**
When creating a public network for a vagrant machine, the flag `libvirt__iface_name` is ignored, and the new network interface is named using the default schema of libvirt (`vnet###`). 

Redacted Vagrantfile contents:
```
Vagrant.configure("2") do |config|
    config.vm.define "test" do |cfg|
        cfg.vm.box = "debian/buster"
        cfg.vm.hostname = "test"

        cfg.vm.network "private_network", model_type: "rtl8139", libvirt__network_name: "default"
        cfg.vm.network "public_network", mode: "bridge", type: "bridge", dev: "br0", mac: "1234567890ab", libvirt__iface_name: "vctest0"
  
        cfg.vm.provider "libvirt" do |lv, override|
            lv.memory = 4096
            lv.cpus = 2
            lv.mgmt_attach = false
            lv.cpu_mode = "host-passthrough"
            override.vm.synced_folder './', '/vagrant', type: 'rsync' 
        end
    end
end
```

The newly created interface should be named `vctest0`, but virsh shows the following:
```bash
virsh # domiflist vagrant_test 
 Interface   Type      Source     Model     MAC
---------------------------------------------------------------
 vnet223     network   default    rtl8139   52:54:00:94:81:bc
 vnet224     bridge    br0        virtio    12:34:56:78:90:ab
```

According to the [docs](https://vagrant-libvirt.github.io/vagrant-libvirt/configuration.html#public-network-options), the option should work for both public and private networks. It seems to be only working for private networks, though.

**To Reproduce**
1. Create a Vagrantfile with a public network
2. Set the option `libvirt__iface_name` to a value of your choice
3. Create the machine using `vagrant up`
4. Verify the interface name by running `virsh domiflist $YOURMACHINE`
5. Realize the interface name is not the same as in your Vagrantfile

**Expected behavior**
The newly created interface should be named as defined in the Vagrantfile.

**Versions (please complete the following information):**:
- Libvirt version: 8.1.0
- Vagrant version [output of `vagrant version`]: 2.2.19
- Vagrant flavour [Upstream or Distro]: Distro
- Vagrant plugins versions (including vagrant-libvirt) [output of `vagrant plugin list`]: 
```
vagrant-libvirt (0.8.2, system)
vagrant-proxyconf (2.0.10, global)
vagrant-reload (0.0.1, global)
```

**Comment by electrofelix** [comment]
You'll need to upgrade vagrant-libvirt, this only appeared in 0.11.2 https://vagrant-libvirt.github.io/vagrant-libvirt/version/0.11.2/configuration.html.

While only started publishing versioned docs during the last year, it was from version 0.10.0, so previous versions have to be looked at via the repo tags https://github.com/vagrant-libvirt/vagrant-libvirt/tree/0.8.2#public-network-options

**Comment by lengers** [comment]
Fair point on the upgrade. I upgraded to 0.11.2, and after a `vagrant up`, my virsh output still looks like this:

````bash
virsh # domiflist vagrant_test 
 Interface   Type      Source     Model    MAC
--------------------------------------------------------------
 vnet236     network   default    virtio   52:54:00:bd:21:4b
 vnet237     bridge    br0        virtio   12:34:56:78:90:ab
```

**Comment by electrofelix** [comment]
Obviously I didn't read the docs correctly myself, it's actually only available in main, yet to be released. https://github.com/vagrant-libvirt/vagrant-libvirt/pull/1692 was the PR that added support (despite the subject saying private, I forgot to correct that to public).

Checked locally and latest main worked as expected
![image](https://user-images.githubusercontent.com/912324/210557796-68d23ece-2fcd-4dc5-ac4a-dc87db600980.png)

If you want to try out the latest you can follow https://github.com/vagrant-libvirt/vagrant-libvirt#latest-development-version to pull down the gem and install locally or add the github packages repo if you are happy with having a token emitted in the plugins.json to provide access.

Going to mark as fixed by #1692 as it is in the latest codebase.
